sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1356,6 +1357,17 @@ def test_asec():
     assert asec(x).rewrite(acsc) == -acsc(x) + pi/2
 
 
+def test_asec_is_real():
+    z = Symbol('z', zero=True)
+    assert asec(z).is_real is None
```

Comment:
Perhaps one could add tests like ``asec(S(1)/2).is_real`` where the values are not numeric.